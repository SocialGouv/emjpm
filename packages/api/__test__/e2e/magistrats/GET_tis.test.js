//

const request = require("supertest");

const { knex } = global;
jest.setMock("@emjpm/api/db/knex", knex);

const server = require("@emjpm/api/app");

const { getTokenByUserType } = require("../utils");

beforeAll(async () => {
  await knex.migrate.latest();
  await knex.seed.run();
});

const simpler = ({ created_at, ...props }) => props;

test("should GET tis for one mandataire ", async () => {
  const token = await getTokenByUserType("mandataire");
  const response = await request(server)
    .get("/api/v1/mandataires/1/tis")
    .set("Authorization", "Bearer " + token);

  expect(simpler(response.body)).toMatchSnapshot();
  expect(response.status).toBe(200);
});
