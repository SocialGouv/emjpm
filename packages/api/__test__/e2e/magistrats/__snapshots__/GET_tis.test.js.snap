// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should GET tis for one mandataire  1`] = `
Object {
  "0": Object {
    "etablissement": "ti arras",
    "id": 1,
    "ti_id": 1,
  },
  "1": Object {
    "etablissement": "ti paris",
    "id": 2,
    "ti_id": 2,
  },
}
`;
